### PR TITLE
Added TC-ACL-2-11 test script

### PR DESCRIPTION
**Pull request by tleacmcsa**
Implement the M-ACL / Managed NIM / ARL test plan.

The test plan (currently in review form) can be found here: 

https://github.com/CHIP-Specifications/chip-test-plans/pull/4316

### CODE REVIEW

**Review by hawk248 — APPROVED** [review]

**Review by robertfarnum — APPROVED** [review]

File: src/python_testing/TC_ACL_2_11.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        asserts.assert_true(isinstance(response, Clusters.AccessControl.Commands.ReviewFabricRestrictionsResponse),
+                            "Result is not of type ReviewFabricRestrictionsResponse")
+
+        self.step(6)
+        WaitForEventReport(arru_queue, Clusters.AccessControl.Events.FabricRestrictionReviewUpdate)
+
+        self.step(7)
+        WaitForEventReport(arec_queue, Clusters.AccessControl.Events.AccessRestrictionEntryChanged)
+
+        self.step(8)
```
Comment (chrisdecenzo) [inline src/python_testing/TC_ACL_2_11.py:162]:
```suggestion
        self.step(7)
```

File: src/python_testing/TC_ACL_2_11.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        subscription_arec.SetEventUpdateCallback(callback=arec_cb)
+        # end
+
+        self.step(5)
+        response = await self.send_single_cmd(cmd=Clusters.AccessControl.Commands.ReviewFabricRestrictions([care_struct]), endpoint=0)
+        asserts.assert_true(isinstance(response, Clusters.AccessControl.Commands.ReviewFabricRestrictionsResponse),
+                            "Result is not of type ReviewFabricRestrictionsResponse")
+
+        self.step(6)
+        WaitForEventReport(arru_queue, Clusters.AccessControl.Events.FabricRestrictionReviewUpdate)
```
Comment (chrisdecenzo) [inline src/python_testing/TC_ACL_2_11.py:157]:
```suggestion
        self.step(6)
        
        logging.info("Please follow instructions provided by the product maker to remove all ARL entries")
                
        WaitForEventReport(arru_queue, Clusters.AccessControl.Events.FabricRestrictionReviewUpdate)
```

**Review by chrisdecenzo — APPROVED** [review]

File: src/python_testing/TC_ACL_2_11.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+    def steps_TC_ACL_2_11(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done"),
+            TestStep(2, "TH1 reads DUT Endpoint 0 AccessControl cluster CommissioningARL attribute"),
+            TestStep(3, "TH1 reads DUT Endpoint 0 AccessControl cluster ARL attribute"),
+            TestStep(4, "For each entry in ARL, iterate over each restriction and attempt access the restriction's ID on the Endpoint and Cluster in the ARL entry.",
+                     "If the restriction is Type AttributeAccessForbidden, read the restriction's attribute ID and verify the response is UNSUPPORTED_ACCESS."
+                     "If the restriction is Type AttributeWriteForbidden, write restriction's the attribute ID and verify the response is UNSUPPORTED_ACCESS."
+                     "If the restriction is Type CommandForbidden, invoke the restriction's command ID and verify the response is UNSUPPORTED_ACCESS."),
```
Comment (tcarmelveilleux) [inline src/python_testing/TC_ACL_2_11.py:78]: This is missing a negative test case that entries that should not be restricted are not restricted.

File: src/python_testing/TC_ACL_2_11.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    def __call__(self, res: EventReadResult, transaction: SubscriptionTransaction):
+        if res.Status == Status.Success and res.Header.ClusterId == self._expected_cluster_id and res.Header.EventId == self._expected_event_id:
+            logging.info(
+                f'Got subscription report for event {self._expected_event_id} on cluster {self._expected_cluster_id}: {res.Data}')
+            self._output.put(res)
+
+
+def WaitForEventReport(q: queue.Queue, expected_event: ClusterEvent):
+    try:
+        res = q.get(block=True, timeout=10)
```
Comment (tcarmelveilleux) [inline src/python_testing/TC_ACL_2_11.py:57]: Test says 1 hour. 10 is OK for CI, but not OK for cert test possibly

**Review by tcarmelveilleux — CHANGES_REQUESTED** [review]
- [ ] Introduce check that ARL doesn't apply to critical elements where it should not apply (e.g. every descriptor across the node, and possibly at least 1 featuremap on every single cluster of the node)
- [ ] Add CI detection argument to avoid long wait for the 1 hour wait

**Review by tcarmelveilleux — APPROVED** [review]
Approved based on follow-ups happening.

**Comment by naveenkommareddi** [comment]
> Approved based on follow-ups happening.

Are the follow-ups tracked on another PR? And will they be for 1.4?

File: src/python_testing/TC_ACL_2_11.py (hunk 3 above)
Comment (chrisdecenzo) [reply src/python_testing/TC_ACL_2_11.py:78]: Will fix in https://github.com/CHIP-Specifications/chip-test-plans/issues/4560

**Review by robertfarnum — APPROVED** [review]

**Comment by tleacmcsa** [comment]
> > Approved based on follow-ups happening.
> 
> Are the follow-ups tracked on another PR? And will they be for 1.4?
> 
> Follow ups:
> 
> * [Managed ACL can block access to critical global attributes CHIP-Specifications/connectedhomeip-spec#10341](https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/10341)
> * [[TC-ACL-2.11] Improve test plan for MACL feature CHIP-Specifications/chip-test-plans#4560](https://github.com/CHIP-Specifications/chip-test-plans/issues/4560)
> * verify subscriptions are not broken when there are ARLs in place

The follow-up test script changes would fall out of updates to the test plan.  Do we need another issue for the test script, or just let the test plan issue cover it all?

**Pull request merged**